src/frontend/js/components/RegisteredAddress/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import type { Address } from 'types/Joanie';
+
+const messages = defineMessages({
+  promoteTitleButton: {
+    id: 'components.AddressesManagement.promoteTitleButton',
+    description: 'Title of the button to set the address as the main one',
+    defaultMessage: 'Define "{title}" address as main',
+  },
+  selectTitleButton: {
+    id: 'components.AddressesManagement.selectTitleButton',
+    description: 'Title of the select button',
+    defaultMessage: 'Select "{title}" address',
+  },
+  selectButton: {
+    id: 'components.AddressesManagement.selectButton',
+    description: 'Label of the select button',
+    defaultMessage: 'Use this address',
+  },
+  editTitleButton: {
+    id: 'components.AddressesManagement.editTitleButton',
+    description: 'Title of the edit button',
+    defaultMessage: 'Edit "{title}" address',
+  },
+  editButton: {
+    id: 'components.AddressesManagement.editButton',
+    description: 'Label of the edit button',
+    defaultMessage: 'Edit',
+  },
+  deleteTitleButton: {
+    id: 'components.AddressesManagement.deleteTitleButton',
+    description: 'Title of the delete button',
+    defaultMessage: 'Delete "{title}" address',
+  },
+  deleteButton: {
+    id: 'components.AddressesManagement.deleteButton',
+    description: 'Label of the delete button',
+    defaultMessage: 'Delete',
+  },
+  actionPromotion: {
+    id: 'components.AddressesManagement.actionPromotion',
+    description: 'Action name for address promotion.',
+    defaultMessage: 'promotion',
+  },
+});
+
+interface Props {
+  address: Address;
+  edit: (address: Address) => void;
+  promote: (address: Address) => void;
+  remove: (address: Address) => void;
+  select: (address: Address) => void;
+}
+
+const RegisteredAddress = ({ promote, select, edit, remove, address }: Props) => {
+  const intl = useIntl();
+
+  return (
+    <li className="registered-addresses-item" key={`addresses-list-${address.id}`}>
+      <button
+        className="button"
+        onClick={() => promote(address)}
+        title={intl.formatMessage(messages.promoteTitleButton, { title: address.title })}
+      >
+        <span
+          className={`address-main-indicator ${
+            address.is_main ? 'address-main-indicator--is-main' : ''
+          }`}
+        />
+      </button>
+      <h6 className="registered-addresses-item__title">{address.title}</h6>
+      <address
+        data-testid={`address-${address.id}-details`}
+        className="registered-addresses-item__address"
+      >
+        {address.first_name}&nbsp;{address.last_name}
+        <br />
+        {address.address} {address.postcode} {address.city}, {address.country}
+      </address>
+      <p className="registered-addresses-item__actions">
+        <button
+          title={intl.formatMessage(messages.selectTitleButton, { title: address.title })}
```

Comment:
I guess this is here mostly for screen readers in order to have a button with more context?

That's a great idea but watch out when using a `title` attribute. For screen reader it's used more as a fallback than anything. So it will not necessarily be announced if the button has content. And if it's announced, it can be announced _in addition_ to the button content. Or not. Behavior is not really consistent between screen readers.

I'd advise to rather use `aria-label` here. `aria-label` will be used as the accessible name of the element, we are sure this will be announced.

If you want to still get a mouse tooltip, I'd leave `title` as is, and add an `aria-label` with the exact same content as the title. For example, NVDA will announce the title in addition to the label, but will announce it only once if it detects it's the same content.

Reply:
Many thanks for this explanation! I will use an aria-label in this case, for mouse user, a tooltip is not necessary in this case.